core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -96,22 +106,65 @@ public BlockLockManager() {}
    * @param blockId the block id
    * @param blockLockType {@link BlockLockType#READ} or {@link BlockLockType#WRITE}
    * @return lock id
+   * @throws FailedToLockBlockException if fails to lock block before timeout
    */
-  public long lockBlock(long sessionId, long blockId, BlockLockType blockLockType) {
-    ClientRWLock blockLock = getBlockLock(blockId);
-    Lock lock;
-    if (blockLockType == BlockLockType.READ) {
-      lock = blockLock.readLock();
-    } else {
-      // Make sure the session isn't already holding the block lock.
-      if (sessionHoldsLock(sessionId, blockId)) {
-        throw new IllegalStateException(String
-            .format("Session %s attempted to take a write lock on block %s, but the session already"
-                + " holds a lock on the block", sessionId, blockId));
+  public long tryLockBlock(long sessionId, long blockId, BlockLockType blockLockType)
+      throws FailedToLockBlockException {
+    Lock lock = getBlockLockWithType(sessionId, blockId, blockLockType);
+    int retryTime = 1;
+    boolean locked = false;
+    try {
+      while (retryTime <  mTryLockRetries) {
+        if (!lock.tryLock(TRY_LOCK_MIN, TimeUnit.MINUTES)) {
+          LOG.debug("Failed in attempt {}/{} to {} lock block of id {} and session {}",
```

Comment:
When my alluxio worker getting low performance, or stuck, I will set this log level to `DEBUG`, and through this log, I still don't know which thread which session hold this block's lock and don't released, is there a way to show the lock holder information?